scalding-serialization/src/main/scala/com/twitter/scalding/serialization/macros/impl/ordered_serialization/providers/TraversablesOrderedBuf.scala

```diff
@@ -120,8 +120,8 @@ object TraversablesOrderedBuf {
     val innerOrd = q"""
       new _root_.scala.math.Ordering[${innerBuf.tpe}] {
         def compare(a: ${innerBuf.tpe}, b: ${innerBuf.tpe}) = {
-          val $ioa = a
-          val $iob = b
+          val $ioa: ${innerBuf.tpe} = a
+          val $iob: ${innerBuf.tpe} = b
```

Comment:
Why do we need these bindings? Why not just pass a and b directly?

Cc @ianoc do you recall?

Reply:
This was around getting the passed variable names into unique ones that are supplied in the macro context. Why we couldn't just put it on the method signature in these cases though is a good question. Maybe a bug with an old version of this/scala or something. It does seem wasteful. This has good test coverage though so I'd be inclined to try collapsing `$ioa` into the `a: ${innerBuf.tpe}` part above?

Reply:
we don't pass a and b themselves I believe since the lower section expands inline to more code (And so having unique names to avoid shadowing of some sort is the goal). But why the args don't just have the names idk

Reply:
ahh yeah, i forgot that aspect of it.